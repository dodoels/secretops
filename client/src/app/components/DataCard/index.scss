.DataCard-container {

  position: relative;
  border-radius: 14px;
  background-color: var(--grey);
  height: 240px;
  max-width: 100%;
  min-width: 240px;
  text-align: center;

  .DataCard-content {
    padding: 10px 20px 0px 20px;
    text-align: left;
    font-weight: bold;

    h3 {
      font-size: 30px;
    }
  }

}
