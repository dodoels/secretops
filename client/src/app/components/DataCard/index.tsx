import './index.scss';
import { Data } from '../../models/Data';

interface DataCardProps {
    data: Data;
}

const DataCard = (props: DataCardProps) => {
    return (
        <div className="DataCard-container">
            <div className="DataCard-content">
                <h3>{props.data.id}</h3>
            </div>
        </div>
    );
};

export default DataCard;
