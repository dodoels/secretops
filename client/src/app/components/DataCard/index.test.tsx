import React from 'react';
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import DataCard from './'; 


describe('<DataCard />', () => {

  it('renders without crash', () => {
    const mockData = { id: 1 };
    render(<DataCard data={mockData} />);
  });

  it('displays the correct data id', () => {
    const mockData = { id: 123 };
    const { getByText } = render(<DataCard data={mockData} />);
    expect(getByText(mockData.id)).toBeDefined();
  });  

  it('matches snapshot', () => {
    const mockData = { id: 1 };
    const { container } = render(<DataCard data={mockData} />);
    expect(container).toMatchSnapshot();
  });

});
