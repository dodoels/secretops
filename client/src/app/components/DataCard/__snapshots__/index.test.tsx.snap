// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DataCard /> > matches snapshot 1`] = `
<div>
  <div
    class="DataCard-container"
  >
    <div
      class="DataCard-content"
    >
      <h3>
        1
      </h3>
    </div>
  </div>
</div>
`;
